src/api/flaskr/service/dashboard/funcs.py

```
@@ -435,6 +440,356 @@ def _load_follow_up_groups_for_progress_record(
+        "source_position": fallback_position,
+        "source_element_bid": "",
+        "source_element_type": "",
+    }
+
+
+def _extract_anchor_element_bid(raw_payload: str) -> str:
+    if not raw_payload:
+        return ""
+    try:
+        payload = json.loads(raw_payload)
+    except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)